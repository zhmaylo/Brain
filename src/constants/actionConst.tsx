/////////////////////////////
//Start. Search section
//
// search string
export const  SEARCH_REQUEST = 'SEARCH_REQUEST';
// 
//End. Search section
/////////////////////////////

/////////////////////////////
//Start. Categories list section
//
// categories list (JSON format)
export const  CATEGORY_LIST = 'CATEGORY_LIST';
//
// view categories list (JSON format)
export const  VIEW_LIST_CAT  = 'VIEW_LIST_CAT';
//
// 
//End. Categories list section
/////////////////////////////


/////////////////////////////
//Start. Filter section
//
// filter all 
export const  MINMAX_DEAL_PRICE = 'MINMAX_DEAL_PRICE';
export const  MIN_SHOW_LIMIT = 'MIN_SHOW_LIMIT';
export const  MAX_SHOW_LIMIT = 'MAX_SHOW_LIMIT';
// 
//End. Filter section
/////////////////////////////


/////////////////////////////
//Start. App init section
//
// application initialization flag
export const IS_APP_INIT = 'IS_APP_INIT';
//
//End. App init section
/////////////////////////////


/////////////////////////////
//Start. Login section
//
// login
export const  LOGIN = 'LOGIN';
// password
export const  PASSWORD = 'PASSWORD';
// 
// passwordMD5
export const  PASS_MD5 = 'PASS_MD5';
// login error
export const  LOGIN_ERROR = 'LOGIN_ERROR';
// 
//End. Login section
/////////////////////////////


/////////////////////////////
//Start. Number column of main screen section
//
// number of columns of displayed products
export const  NUM_COLUM_PROD = 'NUM_COLUM_PROD';
//
//End. Number column of main screen section
/////////////////////////////


/////////////////////////////
//Start. Product current section
//
// product current description
export const  PROD_CURR_DESCRIPTION = 'PROD_CURR_DESCRIPTION';
//
// product current images
export const  PROD_CURR_IMAGES = 'PROD_CURR_IMAGES';
//
//End. Product current section
/////////////////////////////


/////////////////////////////
//Start. Product list section
//
// products list (JSON format)
export const  PRODUCTS_LIST = 'PRODUCTS_LIST';
//
// size view list products
export const  SIZE_LIST_PROD = 'SIZE_LIST_PROD';
//
//End. Product list section
/////////////////////////////


/////////////////////////////
//Start. Progress Bar section
//
// category loading level (for progress bar)
export const  PROG_BAR_MAX_VOL = 'PROG_BAR_MAX_VOL';
export const  PROG_BAR_CUR_VOL = 'PROG_BAR_CUR_VOL';
// 
//End. Progress Bar section
/////////////////////////////


/////////////////////////////
//Start. SID section
//
// SID session 
export const  SESSION_SID  = 'SESSION_SID';
//
//End. SID section
/////////////////////////////


/////////////////////////////
//Start. Sort list products section
//
// ascending sort by price
export const  SORT_PRICE_ASCEND = 'SORT_PRICE_ASCEND';
// descending sort by price
export const  SORT_PRICE_DESCEND = 'SORT_PRICE_DESCEND';
//
// ascending sort by vendor
export const  SORT_VENDOR_ASCEND = 'SORT_VENDOR_ASCEND';
// descending sort by vendor
export const  SORT_VENDOR_DESCEND = 'SORT_VENDOR_DESCEND';
//
// ascending sort by warranty
export const  SORT_WARRANTY_ASCEND = 'SORT_WARRANTY_ASCEND';
// descending sort by warranty
export const  SORT_WARRANTY_DESCEND = 'SORT_WARRANTY_DESCEND';
//
//End. Sort list products section
/////////////////////////////


/////////////////////////////
//Start. Spiner section
//
// spiner toggle (off(false)/ on(true))
export const SPINER_TOGGLE = 'SPINER_TOGGLE';
// 
//End. Spiner section
/////////////////////////////


/////////////////////////////
//Start. Status response section
//
// status response (error/ok)
export const  STATUS_RESPONSE = 'STATUS_RESPONSE';
//
//End. Status response section
/////////////////////////////


/////////////////////////////
//Start. SyncData section
//
// DB work log
export const  DB_LOG = 'DB_LOG';
// 
//End. SyncData section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


/////////////////////////////
//Start.  section
//
//
//End.  section
/////////////////////////////


