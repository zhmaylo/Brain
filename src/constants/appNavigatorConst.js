
/////////////////////////////
// Start. Section 'menu - string values of items'
//
export const MAIN_SCR = "MainScreen";
export const CATEG_SCR = "CategoryScreen";
export const SORT_SCR = "SortScreen";
export const PROD_SCR = "ProdScreen";
export const PROD_GALL_SCR = "ProdGalleryScreen";
export const SYNC_DATA_SCR = "SyncDataScreen";
//
// End. Section 'menu - string values of items'
/////////////////////////////





