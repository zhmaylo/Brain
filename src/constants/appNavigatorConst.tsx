
/////////////////////////////
// Start. Section 'menu - string values of items'
//
export const MAIN_SCR = "MainScreen";
export const CATEG_SCR = "CategoryScreen";
export const SORT_SCR = "SortScreen";
export const PROD_SCR = "ProdScreen";
export const FILTER_SCR = "FilterScreen";
export const SEARCH_SCR = "SearchScreen";
export const PROD_GALL_SCR = "ProdGalleryScreen";
export const SYNC_DATA_SCR = "SyncDataScreen";
export const SETTING_SCR = "SettingScreen";
export const LOGIN_SCR = "LoginScreen";
//
// End. Section 'menu - string values of items'
/////////////////////////////





