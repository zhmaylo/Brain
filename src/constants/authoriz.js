
/////////////////////////////
// Start. Section 'User authorization data'
//
// API_KEY
export const API_KEY = "7408fbfbded5974d9b625d05237f6a6f";
//
// user login/password
export const LOGIN = "dzhmaylo@gmail.com";
export const PASSMD5 = "b4a62dea0279a07036a87e355af638b4";
export const USER_AUTH = 'login=' + LOGIN + '&password=' + PASSMD5;
//
// End. Section 'User authorization data'
/////////////////////////////

