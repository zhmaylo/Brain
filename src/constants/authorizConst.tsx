
/////////////////////////////
// Start. Section 'User authorization data'
//
// 
// 
// user login/password
export const LOGIN_DEFAULT = "dzhmaylo@gmail.com";
let login: string = LOGIN_DEFAULT;
export const setLogin = (loginFromStore: string) => {
    login = loginFromStore;
};

export const PASS_MD5_DEFAULT = "b4a62dea0279a07036a87e355af638b4";
let passMD5: string = PASS_MD5_DEFAULT;
export const setPassMD5 = (PassFromStore: string) => { passMD5 = PassFromStore };

export const USER_AUTH = () => 'login=' + login + '&password=' + passMD5;

//request header for authorization
export const REQUEST_HEADER_AUTH = () => [{
    method: 'POST',
    headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
    body: USER_AUTH()
}];
//
// End. Section 'User authorization data'
/////////////////////////////


/////////////////////////////
// Start. Section 'Time Stamp'
//
// time life session - 5 min = 300000 ms
// export const SID_TIME_LIFE = 300000;
//
// End. Section 'Time Stamp'
/////////////////////////////