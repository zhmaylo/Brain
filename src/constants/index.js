//constants for actions

export const  LOAD_DATA_JSON  = 'LOAD_DATA_JSON'
export const  LOAD_JSON_FLAG  = 'LOAD_JSON_FLAG'
export const  STORAGE_TO_STATE  = 'STORAGE_TO_STATE'



