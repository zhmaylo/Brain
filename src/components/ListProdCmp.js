import React from 'react';
import { View, StyleSheet, FlatList } from 'react-native';
import { WINDOW_WIDTH, WINDOW_HEIGHT } from '../constants/otherConst';
import { BORDER_PAGIN_PROD, STEP_PAGIN_PROD } from '../constants/productsConst';
import { setSizeListProd } from '../api/products/products';
import { ProductCardCmp } from './ProductCardCmp';

let currSize=1;

// view list products
export const ListProdCmp = ({ productList, numCollumns, currSizeList, dispatch }) => {

    // console.log("ListProdCmp.productList => ", productList);
    // console.log("ListProdCmp.numCollumns => ", numCollumns);

    return (
        <View style={styles.container} >
            <FlatList
                numColumns={numCollumns}
                horizontal={false}
                data={productList.slice(0,currSize)}
                renderItem={({ item }) => <ProductCardCmp item={item} numCol={numCollumns} />}
                keyExtractor={item => item.productID + numCollumns}
                key={numCollumns}
                onEndReached={() => {
                    // console.log("start pagination")
                    // console.log("currSizeList, STEP_PAGIN_PROD, productList.length", 
                    //         currSizeList, STEP_PAGIN_PROD, productList.length)
                    
                    currSize = setSizeListProd(currSizeList, STEP_PAGIN_PROD, productList.length);
                    dispatch({ type: 'SIZE_LIST_PROD', payload: currSize });

                    // console.log("currSize pagination => ", currSize);
                }}
                onEndReachedThreshold={BORDER_PAGIN_PROD}
                refreshing={true}
            />
        </View>
    );

}


const styles = StyleSheet.create({
    container: {
        flex: 17,
        resizeMode: "contain",
        width: WINDOW_WIDTH,
        height: WINDOW_HEIGHT,

        backgroundColor: "lightgray",
    }
  
})
