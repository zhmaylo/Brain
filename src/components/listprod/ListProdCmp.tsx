import React from 'react';
import { View, StyleSheet, FlatList, Text } from 'react-native';
import { WINDOW_WIDTH, WINDOW_HEIGHT } from '../../constants/otherConst';
import { STEP_PAGIN_PROD } from '../../constants/productsConst';
import { setSizeListProd } from '../../api/products/products';
import { ProdCardLightCmp } from '../prodcard/ProdCardLightCmp';
import { PROD_EXPECTED } from '../../constants/productsConst';
import { PROD_SCR } from '../../constants/appNavigatorConst';


// current list size 
let currSize = 1;

// view list products
export const ListProdCmp = ({ productList, numCollumns, currSizeList, dispatch, sidAndTime, props }) => {
    // console.log("ListProdCmp.productList => ", productList);
    // console.log("ListProdCmp.numCollumns => ", numCollumns);

    if (productList.length == 0) return (
        <View style={styles.prodNoneCont}>
            <Text style={styles.prodNoneText} > {PROD_EXPECTED} </Text>
        </View>
    )
    else
        return (
            <View style={styles.container} >
                <FlatList
                    numColumns={numCollumns}
                    horizontal={false}
                    data={productList.slice(0, currSize)}
                    // renderItem={({ item }) => console.log('props=> ', props)}
                    renderItem={({ item }) =>
                        (<ProdCardLightCmp item={item}
                            props={props}
                            dispatch={dispatch}
                            sidAndTime={sidAndTime}
                        />)}
                    keyExtractor={item => item.productID + numCollumns}
                    key={numCollumns}
                    onEndReached={() => {
                        // console.log("start pagination")
                        currSize = setSizeListProd(currSizeList, STEP_PAGIN_PROD, productList.length);
                        dispatch({ type: 'SIZE_LIST_PROD', payload: currSize });
                        // console.log("currSize pagination => ", currSize);

                    }}
                    onEndReachedThreshold={2}
                    refreshing={true}



                />
            </View>
        );

}


const styles = StyleSheet.create({
    container: {
        flex: 17,
        resizeMode: "contain",
        width: WINDOW_WIDTH,
        height: WINDOW_HEIGHT,

        backgroundColor: "lightgray",

    },

    prodNoneCont: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: "center",
        alignItems: "center",
    },

    prodNoneText: {
        fontWeight: "700",
        fontSize: 14,
    }

})
