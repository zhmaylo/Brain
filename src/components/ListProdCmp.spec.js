import React from 'react';
import renderer from 'react-test-renderer';
import { ListProdCmp } from './ListProdCmp';
import {PRODUCTS_FROM_FILE} from '../constants/productsJSON';


let productsList = PRODUCTS_FROM_FILE[0].result.list;
console.log("ListProdCmp.spec => productsList", PRODUCTS_FROM_FILE[0].result.list[0]);

const callback = () => {
    console.log("Test - callback test  ");
    // expect(data).toMatch('Test Ok');
    return;
};

test('renders correctly', () => {
    
    const tree = renderer.create(
        ListProdCmp({
                productList : productsList,
                numCollumns : 1,
                currSizeList : 20,
                dispatch : callback }
        ))
        // ListProdCmp ({productsList, 1, 20, callback}))
        // ListProdCmp (productsList, 1, 20, callback))
        // ListProdCmp ({productsList, 1}))
        .toJSON();
    expect(tree).toMatchSnapshot();
});

   


// tasca - testAsyncCallback Creates Async Callback test
// taspr - testAsyncPromise Creates Aysnc Promise test
// tasaw - testAsyncAwait Creates Async Await test
// tsnsh - testSnapShot Creates Snapshot test
// tshre - testShallowRenderer Creates Shallow Renderer test