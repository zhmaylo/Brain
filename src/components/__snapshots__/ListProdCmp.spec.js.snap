// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "lightgray",
      "flex": 17,
      "height": 1334,
      "resizeMode": "contain",
      "width": 750,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "EAN": "",
          "FOP": 0,
          "articul": "SV300S37A/120G",
          "available": Array [],
          "bonus": 0,
          "brief_description": "внутренний, V300, Тип флеш-памяти - MLC, SATA 6Gb/s, 450 Mb/s, 450 Mb/s, 1м. ч.",
          "categoryID": "1235",
          "country": "США",
          "full_image": "http://brain.com.ua/static/common/images/no-photo-api.png",
          "is_archive": "0",
          "is_exclusive": "0",
          "is_new": "0",
          "kbt": "0",
          "large_image": "http://opt.brain.com.ua/static/images/prod_img/4/1/U0033541_big.jpg",
          "medium_image": "http://opt.brain.com.ua/static/images/prod_img/4/1/U0033541.jpg",
          "name": "Накопитель SSD 2.5\\" 120GB Kingston (SV300S37A/120G)",
          "prepayment_amount": "0",
          "price": "23.05",
          "price_uah": "637.00",
          "productID": "41681",
          "product_code": "U0033541",
          "quantity_package_sale": 0,
          "recommendable_price": "0.00",
          "retail_price_uah": "666.00",
          "small_image": "http://opt.brain.com.ua/static/images/prod_img/4/1/U0033541_small.jpg",
          "stocks": Array [],
          "stocks_expected": Array [],
          "vendorID": "25",
          "volume": "0",
          "warranty": "36",
          "weight": "0.15",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-around",
              "margin": 1,
              "padding": 5,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View>
              <Image
                source={
                  Object {
                    "uri": "http://opt.brain.com.ua/static/images/prod_img/4/1/U0033541_big.jpg",
                  }
                }
                style={
                  Object {
                    "height": 266.8,
                    "resizeMode": "contain",
                  }
                }
              />
              <Text
                style={
                  Object {
                    "fontSize": 14,
                    "paddingHorizontal": 5,
                  }
                }
              >
                Накопитель SSD 2.5" 120GB Kingston (SV300S37A/120G)
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "700",
                    "paddingHorizontal": 5,
                  }
                }
              >
                666.00
                 грн 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                    "paddingHorizontal": 5,
                  }
                }
              >
                Гарантия: 
                36
                мec.
                 
                Код: 
                U0033541
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                    "paddingHorizontal": 5,
                  }
                }
              >
                ID: 
                41681
                 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
